src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,90 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [descriptionField, setDescriptionField] = useState('');
+  const [titleField, setTitleField] = useState('');
+  const [imgUrlField, setImgUrlField] = useState('');
+  const [imdbUrlField, setImdbUrlField] = useState('');
+  const [imdbIdField, setImdbIdField] = useState('');
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const movieToUpload = {
+      title: titleField,
+      description: descriptionField,
+      imgUrl: imgUrlField,
+      imdbUrl: imdbUrlField,
+      imdbId: imdbIdField,
+    };
+
+    const reset = () => {
+      setDescriptionField('');
+      setTitleField('');
+      setImgUrlField('');
+      setImdbUrlField('');
+      setImdbIdField('');
+    };
+
+    onAdd(movieToUpload);
+    setCount((current) => current + 1);
```

Comment:
here you could just write like this
```suggestion
    setCount(count + 1);
```